cmd/chartsvc/handler.go

```diff
@@ -200,6 +200,28 @@ func getChartVersionValues(w http.ResponseWriter, req *http.Request, params Para
 	w.Write([]byte(files.Values))
 }
 
+// listChartsWithFilters returns the list of repos that contains the given chart and the latest version found
+func listChartsWithFilters(w http.ResponseWriter, req *http.Request, params Params) {
+	db, closer := dbSession.DB()
+	defer closer()
+
+	var charts []*models.Chart
+	if err := db.C(chartCollection).Find(bson.M{
+		"name": params["chartName"],
+		"chartversions": bson.M{
+			"$elemMatch": bson.M{"version": req.FormValue("version"), "appversion": req.FormValue("appversion")},
+		}}).Select(bson.M{
+		"name": 1, "repo": 1,
+		"chartversions": bson.M{"$slice": 1},
+	}).All(&charts); err != nil {
+		log.WithError(err).Errorf("could not find the given chart with id %s", params["chartName"])
+		// continue to return empty list
+	}
+
+	cl := newChartListResponse(charts)
```

Comment:
This helper will filter out repos if the latest version is a duplicate from a different repo. I think this is okay, because if the `bitnami` and `stable` repo have the same latest versions, it'll just show one of them. However, if one of them had a newer version that the other, it will return both since the latest versions are not duplicates.

Just double checking since you're more aware with how this fits in with the other changes and requirements, does this work the way we want?

Reply:
Yes, I am aware of that behavior. I think it's okay to return just one chart (from a single repo) in that case since we are just reporting that the version exists.